client/cli/src/main.rs

```diff
@@ -103,6 +104,11 @@ enum CliOutput {
         connect_info: WsClientConnectInfo,
     },
 
+    DecodeConnectInfo {
+        urls: Vec<Url>,
+        id: FederationId,
```

Comment:
Maybe this can just wrap the `WsClientConnectInfo` transparently? Not sure if that would need [additional annotation](https://serde.rs/attr-flatten.html) to work or if `DecodeConnectInfo(WsClientConnectInfo)` would do the trick.

Reply:
My attempts along these lines don't seem to work as they always output the bech32 string. Maybe I need to add something to `WsClientConnectInfo` so it can output its JSON representation directly?

Reply:
Ah, we probably defined a custom serde serialization, then keep it as is :)